### PR TITLE
Added test for convertPlaceholder regex

### PR DESCRIPTION
**Pull request by WhiteOlivierus**
As promised in #461

### CODE REVIEW

File: extension/task/utils/parseConfigFile.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import * as path from "path";
 import * as tl from "azure-pipelines-task-lib/task";
 import { getVariable } from "azure-pipelines-task-lib/task";
 import { ISharedVariables } from "./getSharedVariables";
-import convertPlaceholder from "./convertPlaceholder";
+import {convertPlaceholder} from "./convertPlaceholder";
```
Comment (mburumaxwell) [inline extension/task/utils/parseConfigFile.ts:12]:
```suggestion
import { convertPlaceholder } from "./convertPlaceholder";
```

File: extension/package.json [hunk 2]
```diff
@@ -35,7 +35,8 @@
     "@types/node": "^18.11.18",
     "@types/q": "^1.5.5",
     "jest": "^29.3.1",
-    "ts-jest": "^29.0.5",
-    "typescript": "^4.4.3"
+    "ts-jest": "^29.0.3",
```
Comment (mburumaxwell) [inline extension/package.json:38]:
```suggestion
    "ts-jest": "^29.0.5",
```

File: extension/task/utils/convertPlaceholder.ts [hunk 3]
```diff
@@ -15,8 +12,23 @@ export default function convertPlaceholder(input: string): string {
   return result;
 }
 
-function matchAll(target: string, rExp: RegExp, matches: Array<RegExpExecArray> = []) {
+function extractPlaceholder(input: string) {
+  const regexp: RegExp = new RegExp(
+    "\\${{\\s*([a-zA-Z_]+[a-zA-Z0-9_-]*)\\s*}}",
+    "g"
+  );
```
Comment (mburumaxwell) [inline extension/task/utils/convertPlaceholder.ts:19]:
```suggestion
  const regexp: RegExp = new RegExp("\\${{\\s*([a-zA-Z_]+[a-zA-Z0-9_-]*)\\s*}}", 'g');
```

**Review by mburumaxwell — COMMENTED** [review]
I have made some suggestions you can have a look.

How about testing two extra scenarios?
1. Without the `PAT:` prefix
2. With malformed brackets
3. Without the relevant variable supplied

File: extension/package.json (hunk 2 above)
Comment (WhiteOlivierus) [reply extension/package.json:38]: Is this a version that is locked in?

Comment (mburumaxwell) [reply extension/package.json:38]: No but I don't see why it should be downgraded

**Pull request merged**